code/controllers/TaxonomyDirectoryController.php

```diff
@@ -14,22 +14,32 @@ class TaxonomyDirectoryController extends Page_Controller
 
     public function index(SS_HTTPRequest $HTTPRequest)
     {
-        $termString = $HTTPRequest->param('ID');
-
-        $pages = Page::get()
-            ->innerJoin(
-                'BasePage_Terms',
-                '"Page"."ID"="BasePage_Terms"."BasePageID"')
-            ->innerJoin(
-                'TaxonomyTerm',
-                "\"BasePage_Terms\".\"TaxonomyTermID\"=\"TaxonomyTerm\".\"ID\" AND \"TaxonomyTerm\".\"Name\" = '$termString'");
-
-        return $this->customise(new ArrayData(array(
-            'Title' => $termString,
-            'Term' => $termString,
-            'Pages' => $pages,
-            'Breadcrumbs' => $this->renderBreadcrumb($termString)
-        )))->renderWith(array("TaxonomyDirectory", "Page"));
+        //look up the term to avoid sql injection by using the ORM
+        $term = TaxonomyTerm::get()->filter(['URLSegment' => $HTTPRequest->param('ID')])->first();
+
+        if ($term) {
+            $data = ArrayData::create(array(
+                'Title' => $term->Name,
+                'Results' => Page::get()
```

Comment:
This is an ORM call, based on the docs example. Works as expected. Feel free to suggest a simplification of this here if it's highly desirable otherwise good to go.

Reply:
To that request; dot notation would be greatly preferable if you can swing it
e.g. `filter(['thing.relationship.property' => $value])`

Reply:
I think it should be:

```php
Page::get()->filter([
    'Terms.ID' => $term->ID,
]);
```

But hang on a second, where is this `many_many` declared? Where is `BasePage`/`BasePage_Terms` coming from?

Reply:
BasePage is a CWP thing (making an assumption), which shouldn't be directly referenced (tight coupling is less than ideal).

https://github.com/silverstripe/cwp/blob/master/src/PageTypes/BasePage.php

This module isn't CWP specific, nor is CWP a dependency.

Reply:
Is there a way to implement this so that the feature we're trying to get here (proper URLSegment slug access to filtered content based on a taxonomy tag) while both keeping it generic AND it working within a CWP implementation? Any practical approaches to recommend here?

On CWP Terms are pinned to BasePage [here](https://github.com/silverstripe/cwp/blob/1.8/code/extensions/TaxonomyTermExtension.php), in a generic implementation they could be pinned to any page type (but most likely Page). Is there an ORM call that would return correctly in both those examples?

Reply:
A service definition via Injector that defines the relevant base classname. By default it could be `Page` and a header with _after: #first-part; only: module exists cwp/cwp_ can be added to change that to `CWP\CWP\BasePage` (caveat: I've not checked anything in this post, it's pure pseudo code, but I hope the concept is demonstrated all the same).